src/main/java/org/myrobotlab/service/Adafruit16CServoDriver.java

```diff
@@ -1041,5 +1043,28 @@ public String getBus() {
   public String getAddress() {
     return deviceAddress;
   }
+  
+  @Override
+  public void startService() {
+    super.startService();
+    registerForInterfaceChange(I2CController.class);
+  }
+  
+  @Override
+  public void onInterfaceRegistered(String serviceName, String interfaceName) {
+    if (I2CController.class.toString().equals(interfaceName)) {
+      controllers.add(serviceName);

```

Comment:
I'm not sure I like the idea that we're tracking a set of all possible motor controllers here.  It seems like the real intention of this code is only to provide a helper to the webgui.  Wouldn't it be easier if runtime maintained the interface type to service name list ?  Upon initialization, this service needs to initialize the controllers list. o/w it could get out of sync with the truth that is owned by runtime.

Reply:
runtime owns the data now - and the webgui now processes the single compressed message of changes to the runtime data structure requestedAttachMatrix

Yeah it is a helper function ... for "any" gui .. but I realize its a pretty important concept.
With Java reflection (or any language that support self inspection of a class) - a system can determine what a service "provides" ... but none of them can provide - what a service is "looking for"
So creating these methods and data structure - gives opportunity for services to publish what they are interested in and examine what services can provide it.  The concept goes beyond just UI support.